int32 state_x
int32 state_y
int32 count
